animations/lib/src/basics/09_animated_switcher.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+double generateHeight() => Random().nextDouble() * 200;
+
+double generateWidth() => Random().nextDouble() * 200;
+
+Widget generateContainer() => Container(
+      key: GlobalKey(),
```

Comment:
GlobalKeys are particularly expensive to use. Try having a field that keeps count of how many widgets have been built, and then use ValueKey, with the value of the counter for each one. That'll keep them unique and avoid the expense of a GlobalKey.